### PR TITLE
Receive TX

### PR DESCRIPTION
**Pull request by shargon**
image

### CODE REVIEW

File: src/NeoSharp.Core/Network/Server.cs [hunk 1]
```diff
@@ -131,9 +131,11 @@ public void ConnectToPeers(params EndPoint[] endPoints)
         #region IBroadcaster implementation
 
         /// <inheritdoc />
-        public void Broadcast(Message message, IPeer source)
+        public Task Broadcast(Message message, IPeer source)
```
Comment (osmirnov) [inline src/NeoSharp.Core/Network/Server.cs]: What's the point to return always completed task?

File: src/NeoSharp.Core/Network/Server.cs [hunk 2]
```diff
@@ -131,9 +131,9 @@ public void ConnectToPeers(params EndPoint[] endPoints)
         #region IBroadcaster implementation
 
         /// <inheritdoc />
-        public void Broadcast(Message message, IPeer source)
+        public async Task Broadcast(Message message, IPeer source)
         {
-            Parallel.ForEach(_connectedPeers.Where(p => p != source), p => p.Send(message));
+            await Task.Factory.StartNew(() => Parallel.ForEach(_connectedPeers.Where(p => p != source), p=> p.Send(message)));
```
Comment (osmirnov) [inline src/NeoSharp.Core/Network/Server.cs:136]: Once again, it is the same as Task.CompleteTask and we do not need to await on broadcasting.

Comment (shargon) [reply src/NeoSharp.Core/Network/Server.cs:136]: So what is the right way to do this in async way?

Comment (osmirnov) [reply src/NeoSharp.Core/Network/Server.cs:136]: Send is async method. It does not block the thread. Parallel.ForEach makes sure Send is called for all peers at the same time. My idea was we don't want to wait until broadcasting will be completed. That is why we call this (async) and do not wait until all these operations will be done. The void result prevents from awaiting and blocking the current thread.

**Comment by shargon** [comment]
Could you add this change?

**Pull request merged**